docs/design/ocs-to-ocs-storageconsumer.md

```diff
@@ -0,0 +1,46 @@
+# OCS to OCSs : StorageConsumer CRD
+
+## Introduction
+This document describes the design of StorageConsumer CR(and accompanying controller) representing consumer cluster inside the provider cluster. An abstraction on top of lower level rook resources that need to be defined in order to enable storage consumption by a specific consumer cluster and acts as a source of truth for consumer cluster requests/configuration. 
+
+## Design Details
+
+### API design of StorageConsumer
+```yaml
+apiVersion: ocs.openshift.io/v1alpha1
+kind: StorageConsumer
+metadata:
+  name: consumer-1
+  finalizers:
+    - <string>
+  namespace: openshift-storage
+spec:
+  capacity: <resource.Quantity>
+status:
+  state: <string>
+  grantedCapacity: <resource.Quantity>
+  cephObjects:  
+    blockPoolName: <string>
+    cephUser: <string>
+  connectionDetails: <*v1.SecretKeySelector>
+```
+
+* `spec.capacity` Will define the desired capacity (quota)
+requested by a consumer cluster deployment.
+
+* `status.grantedCapacity` Will report the capacity granted to the consumer cluster by the provider cluster
+
+* `status.state` Will report the state of the StorageConsumer. Available states are Ready, Configuring, Deleting, Failed. 
+
+* `status.cephObjects` Will report the created ceph objects required for external storage. 
+
+* `status.connectionDetails` Will hold the reference to secret containing external connection details (blob). 
```

Comment:
Add a `TODO` item to provide the definition/structure of the Secrets we'll be referencing.